### PR TITLE
save/load model from s3

### LINKED ISSUE #993: Add function to load/save s3 model
Add functionality to load/save model from/to Amazon s3 files in BigDL.

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?

save/load model from s3

## How was this patch tested?

unit tests, integration tests

## Related links or issues (optional)
fixed #993

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/File.scala [hunk 1]
```diff
@@ -53,31 +54,56 @@ object File {
   }
 
   /**
-   * Save scala object into a local/hdfs path
+   * Save scala object into a local/hdfs/s3 path
+   *
+   * Notice: S3 path should be like s3a://bucket/xxx.
+   *
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/File.scala]: check the format of s3 path, give error message if not correct

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/File.scala [hunk 2]
```diff
@@ -53,31 +54,56 @@ object File {
...
+          throw new RuntimeException(s"file $fileName already exists")
+        }
       }
+      out = fs.create(dest)
+      val byteArrayOut = new ByteArrayOutputStream()
+      objFile = new ObjectOutputStream(byteArrayOut)
+      objFile.writeObject(obj)
+      IOUtils.copyBytes(new ByteArrayInputStream(byteArrayOut.toByteArray), out, 1024, true)
+      objFile.close()
+      out.close()
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/File.scala]: objFile.close()
out.close()

have been done in finally

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/File.scala [hunk 3]
```diff
@@ -136,29 +162,46 @@ object File {
...
+    var in: FSDataInputStream = null
+    var objFile: ObjectInputStream = null
+    try {
+      fs = src.getFileSystem(conf)
+      in = fs.open(src)
+      val byteArrayOut = new ByteArrayOutputStream()
+      IOUtils.copyBytes(in, byteArrayOut, 1024, true)
+      in.close()
+      fs.close()
+      objFile = new ObjectInputStream(new ByteArrayInputStream(byteArrayOut.toByteArray))
```
Comment (SeaOfOcean) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/File.scala]: in.close()
 fs.close()

done in finally

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/integration/HdfsSpec.scala [hunk 4]
```diff
@@ -106,7 +106,7 @@ class HdfsSpec extends FlatSpec with Matchers with BeforeAndAfter{
 
     def writeToHdfs(localFile: String, hdfsDir: String): Unit = {
       val src = new Path(localFile)
-      val fs = src.getFileSystem(new Configuration())
+      val fs = src.getFileSystem(new Configuration(false))
```
Comment (zhangxiaoli73) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/integration/HdfsSpec.scala]: Why did this unit test pass before?

File: pom.xml [hunk 5]
```diff
@@ -229,6 +229,12 @@
             </dependency>
             <dependency>
                 <groupId>org.apache.hadoop</groupId>
+                <artifactId>hadoop-aws</artifactId>
```
Comment (zhangxiaoli73) [inline pom.xml]: have to add here?

Comment (qiuxin2012) [inline pom.xml]: yes

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/integration/HdfsSpec.scala (hunk 4 above)
Comment (qiuxin2012) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/integration/HdfsSpec.scala]: We didn't configure default fs in jenkins before.

**Comment by qiuxin2012** [comment]
jenkins passed

**Pull request merged**